File: 1392-find-the-difference-of-two-arrays/1392-find-the-difference-of-two-arrays.java
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        Set<Integer> uniqueNums1 = convertToSet(nums1);
        Set<Integer> uniqueNums2 = convertToSet(nums2);

        List<List<Integer>> diffs = new ArrayList<>();
        diffs.add(new ArrayList<Integer>());
        diffs.add(new ArrayList<Integer>());

        for(int num : uniqueNums1){
            if(!uniqueNums2.contains(num)){
                diffs.get(0).add(num);
            }
        }

        for(int num : uniqueNums2){
            if(!uniqueNums1.contains(num)){
                diffs.get(1).add(num);
            }
        }

        return diffs;
    }

    private Set<Integer> convertToSet(int[] nums1) {
        Set<Integer> set = new HashSet<>();
        for (int val : nums1)
            set.add(val);
        return set;
    }
}

File: 1392-find-the-difference-of-two-arrays/1392-find-the-difference-of-two-arrays.kt
class Solution {
    fun findDifference(nums1: IntArray, nums2: IntArray): List<List<Int>> {

        val diffNums = mutableListOf<MutableList<Int>>()
        diffNums.add(mutableListOf())
        diffNums.add(mutableListOf())

        val uniqueNum1 = nums1.toSet()
        val uniqueNum2 = nums2.toSet()

        for(num in uniqueNum1){

            if(!uniqueNum2.contains(num)){
                diffNums[0].add(num) 
            }
        }

        for(num in uniqueNum2){

            if(!uniqueNum1.contains(num)){
                  diffNums[1].add(num) 
            }
        }
        return diffNums
    }
}